homeassistant/components/cover/mqtt.py

```diff
@@ -67,30 +92,48 @@ def async_setup_platform(hass, config, async_add_devices, discovery_info=None):
         config.get(CONF_PAYLOAD_STOP),
         config.get(CONF_OPTIMISTIC),
         value_template,
+        config.get(CONF_TILT_OPEN_POSITION),
+        config.get(CONF_TILT_CLOSED_POSITION),
+        config.get(CONF_DEVICE_CLASS),
+        config.get(CONF_TILT_MIN),
+        config.get(CONF_TILT_MAX),
     )])
 
 
 class MqttCover(CoverDevice):
     """Representation of a cover that can be controlled using MQTT."""
 
-    def __init__(self, name, state_topic, command_topic, qos, retain,
-                 state_open, state_closed, payload_open, payload_close,
-                 payload_stop, optimistic, value_template):
+    def __init__(self, name, state_topic, command_topic, tilt_command_topic,
+                 tilt_status_topic, qos, retain, state_open, state_closed,
+                 payload_open, payload_close, payload_stop,
+                 optimistic, value_template, tilt_open_position,
+                 tilt_closed_position, device_class, tilt_min, tilt_max):
         """Initialize the cover."""
+        if tilt_command_topic is not None:
+            _LOGGER.info("MQTT cover configured with tilt topic: "
+                         + tilt_command_topic)
         self._position = None
         self._state = None
         self._name = name
         self._state_topic = state_topic
         self._command_topic = command_topic
+        self._tilt_command_topic = tilt_command_topic
+        self._tilt_status_topic = tilt_status_topic
         self._qos = qos
         self._payload_open = payload_open
         self._payload_close = payload_close
         self._payload_stop = payload_stop
         self._state_open = state_open
         self._state_closed = state_closed
         self._retain = retain
+        self._tilt_open_position = tilt_open_position
+        self._tilt_closed_position = tilt_closed_position
         self._optimistic = optimistic or state_topic is None
         self._template = value_template
+        self._device_class = device_class
+        self._tilt_value = STATE_UNKNOWN
```

Comment:
if _tilt_value is set to None you'll not be able to change it through the UI because it's not there in supported_features